config/openshift/v4_12_exp/translate.go

```diff
@@ -295,3 +297,28 @@ func validateMCOSupport(mc result.MachineConfig, ts translate.TranslationSet) re
 	}
 	return cutil.TranslateReportPaths(r, ts)
 }
+
+// fcos v1.5.0-exp generates a user.cfg file using append; however, OpenShift config
+// does not support append (since MCO does not support it). Let change the file to use contents
+func translateUserGrubCfg(config *types.Config, ts *translate.TranslationSet) translate.TranslationSet {
+	newMappings := translate.NewTranslationSet("json", "json")
+	for i, file := range config.Storage.Files {
+		if file.Path == "/boot/grub2/user.cfg" {
+			if len(file.Append) == 1 {
+				translatedPath := path.New("json", "storage", "files", i, "contents")
+				config.Storage.Files[i].FileEmbedded1.Contents.Source = file.Append[0].Source
+				newMappings.AddTranslation(path.New("json", "storage", "files", i, "append", 0, "source"), translatedPath.Append("source"))
+				config.Storage.Files[i].FileEmbedded1.Contents.Compression = file.Append[0].Compression
+				newMappings.AddTranslation(path.New("json", "storage", "files", i, "append", 0, "compression"), translatedPath.Append("compression"))
+				config.Storage.Files[i].FileEmbedded1.Append = nil
+
+				newMappings.AddTranslation(path.New("json", "storage", "files", i, "append", 0), translatedPath)
+			} else {
+				panic(fmt.Errorf("translating user.cfg: expected one append object but instead %v were found", len(config.Storage.Files)))
```

Comment:
It's possible for a user to have (improperly) done this by hand:

```yaml
variant: openshift
version: 4.12.0-experimental
metadata:
  name: foo
  labels:
    machineconfiguration.openshift.io/role: worker
storage:
  files:
    - path: /boot/grub2/user.cfg
      append:
        - inline: foo
        - inline: bar
```

so we can't panic here because that's reserved for programmer errors.  The existing validation code will notice and reject the `append` section later, so I think all we have to do here is return early.